tests/leak-tests/pom.xml

```diff
@@ -0,0 +1,488 @@
+<!--
```

Comment:
I will remove the test-jar


My intention was to be equivalent to integration-tests.. that's why I copied the dependencies here.

Reply:
It seems unlikely folks will keep the two places in line going forward, so I dont see much reason for them to start that way when most of the dependencies arent even used here. We should just add dependencies as they are actually needed and keep the tests closer to the way things are actually used in practice in e.g the broker distribution.

Reply:
Unresolving comment since the test jar is still being created. Plus the 'entire pom is mostly unused stuff' aspect also still stands.

Reply:
I am only keeping MQTT Paho client for now...other than that... I think it's all clear...

Reply:
There still appears to be rather a lot kept around that are either unused or superfluous, even besides the MQTT 3 and 5 clients (which if they arent used as it seems, can be dropped added whenever any tests are added that use them).

Most of the dependencies still seem unused or superfluous repetition of transitive deps...e.g various individual netty deps? They seem like they should all be implicit transitively unless the tests are explicitly using netty themslves, which they dont appear to be. Is anything using artemis-ra, artemis-spring-integration, transaction and management specs? Etc.

Reply:
I removed all I could.


I could still remove stomp-server and MQTT server, however I don't want to do that.. as I want to keep these libraries in the broker... not just for future usage, but also to keep the broker's protocols closer to the runtime.

Reply:
I will have to add a few back (Probably JSON only.. but I'm reopening this until I fix it)

Reply:
@gemmellr fixed JSON. At this point I don't know what else to take out.

Reply:
Commented on some more that seem like they could go. I can live with the broker protocol dependencies since that is a representative-test thing.

Not being able to easily look at whats there and see something is unused or unnecessary to declare (i.e purely transitive) as youve found, is precisely why it should start with the minimal appropriate and then only have things added/removed as we go.